src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -389,6 +398,10 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
             }
         }
 
+        // TODO: Let the OS (or SAM CLI) assign the port, then we need to
+        // scrape SAM CLI to find the port that was actually used?
+        const apiPort = config.invokeTarget.target === 'api' ? await getStartPort() : undefined
+        const debugPort = config.noDebug ? undefined : (apiPort ?? (await getStartPort())) + 1
```

Comment:
I'm thinking this is a bad idea...`getPortPromise` increments the port number until it finds an open port, and incrementing that manually without another `getPortPromise` check means that there's a change you'll run into a clash since you don't have anything inspecting whether or not the next port is valid.

Thinking you'll want to modify `getStartPort` to something like:
```
export async function getStartPort(portNumber: number = 5858): Promise<number> {
    // should we let the user configure the starting port?
    return getPortPromise({ port: portNumber })
}
```

and modify this call to `await getStartPort(apiPort + 1)`

Reply:
good call, fixed in 54e9898948ef